src/lib/dispatch.ts

```diff
@@ -10,3 +10,16 @@ export const dispatcher = () => {
     detail,
   }));
 };
+
+
+export const dispatcherWithEventPassThrough = () => {
+  const element = get_current_component() as HTMLElement;
+  return <EventType extends Event>(name: string, event: EventType) => {
+    const copyEvent = new Proxy(event, 
```

Comment:
I'm slightly worried about overall performance of adding proxy objects for every button in the app, so if we can eventually find a solution that doesn't require proxies I think I'd want to switch to it. Maybe adding an event listener whenever the button becomes disabled that cancels the event and stops propagation would work 🤔. But I think this is great for now! Good stuff.

Reply:
I can probably add a flag that makes this functionality optional - didn't realize Proxies would have that much of a performance hit

Reply:
yeah they're in the class of avoid unless necessary for me. They're apparently quite hard to optimize when compiling :( https://gomakethings.com/proxies-vs.-setter-functions-in-state-based-ui-libraries/ and the concern here is cumulative since we add one for every button.